### ISSUE TITLE
Scikit-learn classification

### ISSUE THREAD

**Issue by viggo-gascou** [viggo-gascou/fyp2022p03GroupB2#7]
Run all classification experiments on both of the tasks you've chosen (one binary and one
multi-class task). 
Evaluate your different classifiers on the validation set and report relevant evaluation metrics 
(_accuracy, precision/recall/F-score_).

As a baseline, start with the `sklearn. linear _model.SGDClassifier` in a logistic regression configuration (`loss='10g'`) using bag of words features. 

Then run additional experiments trying to improve your initial scores by any means you can think of. Try out at least
4 different methods. 

Usually you will need to run several experiments for each methods to test different parameter values. Here are some settings in scikit-learn that you could experiment with:

* Additional preprocessing options: n-gram features, lowercasing, stop word lists (see
options to sklearn. `feature_extraction.text` `CountVectorizer`.
* Count transformations (`sklearn.feature extraction.text.TfIdfTransformer`).
* The classification loss (loss parameter to `SGDClassifier`).
* The regularisation strength (alpha parameter to `SGDClassifier` - try varying it in exponentially spaced steps).
* Different classifiers (e.g., `sklearn.ensemble.RandomForestClassifier` or `sklearn.naive_bayes.MultinomialNB`
* Anything else discussed during the lessons, or implemented in `scikit-learn`.

For the systems that achieves the highest accuracy on the validation set, run the evaluation on the test set and report your results.
